--- nordicsdk/libraries/cli/nrf_cli.h	2019-05-05 11:34:26.310763346 -0400
+++ nordicsdk/libraries/cli/nrf_cli.h	2019-05-05 11:34:32.277400766 -0400
@@ -209,7 +209,8 @@
 {
     NRF_CLI_RECEIVE_DEFAULT,
     NRF_CLI_RECEIVE_ESC,
-    NRF_CLI_RECEIVE_ESC_SEQ
+    NRF_CLI_RECEIVE_ESC_SEQ,
+    NRF_CLI_RECEIVE_TILDE,
 } nrf_cli_receive_t;
 
 
--- nordicsdk/libraries/cli/nrf_cli.c	2019-05-05 11:17:23.984605522 -0400
+++ nordicsdk/libraries/cli/nrf_cli.c	2019-05-05 11:33:35.747618072 -0400
@@ -1733,7 +1733,7 @@
                     case NRF_CLI_VT100_ASCII_DEL:       /* DELETE */
                         if (NRF_CLI_ECHO_ON_MODE(p_cli))
                         {
-                            char_delete(p_cli);
+                            char_backspace(p_cli);
                         }
                         break;
                     default:
@@ -1794,10 +1794,17 @@
                     case 'L': /* INSERT Button */
                         p_cli->p_ctx->insert_mode = !p_cli->p_ctx->insert_mode;
                         break;
+                    case '3': /* DELETE Button */
+                    	char_delete(p_cli);
+                    	NRF_CLI_RECIEVE_STATE_NEXT(p_cli, NRF_CLI_RECEIVE_TILDE);
+                    	break;
                     default:
                         break;
                 }
                 break;
+			case NRF_CLI_RECEIVE_TILDE:
+				NRF_CLI_RECIEVE_STATE_NEXT(p_cli, NRF_CLI_RECEIVE_DEFAULT);
+				break;
             default:
                 NRF_CLI_RECIEVE_STATE_NEXT(p_cli, NRF_CLI_RECEIVE_DEFAULT);
                 break;
