--- nordicsdk/libraries/experimental_ringbuf/nrf_ringbuf.c	2019-03-31 16:21:21.777055349 -0400
+++ nordicsdk/libraries/experimental_ringbuf/nrf_ringbuf.c	2019-03-31 16:21:38.730307633 -0400
@@ -80,7 +80,7 @@
     uint32_t wr_idx = p_ringbuf->p_cb->tmp_wr_idx & p_ringbuf->bufsize_mask;
     uint32_t rd_idx = p_ringbuf->p_cb->rd_idx & p_ringbuf->bufsize_mask;
     uint32_t available = (wr_idx >= rd_idx) ? p_ringbuf->bufsize_mask + 1 - wr_idx :
-            p_ringbuf->p_cb->rd_idx -  p_ringbuf->p_cb->tmp_wr_idx;
+            p_ringbuf->p_cb->rd_idx -  (p_ringbuf->p_cb->tmp_wr_idx - (p_ringbuf->bufsize_mask + 1));
     *p_length = *p_length < available ? *p_length : available;
 
     *pp_data = &p_ringbuf->p_buffer[wr_idx];
@@ -209,7 +209,7 @@
 
     if (length > trail)
     {
-        memcpy(&p_ringbuf->p_buffer[masked_rd_idx], p_data, trail);
+        memcpy(p_data, &p_ringbuf->p_buffer[masked_rd_idx], trail);
         length -= trail;
         masked_rd_idx = 0;
         p_data += trail;
