--- ./nordicsdk/drivers_nrf/uart/nrf_drv_uart.c	2019-04-07 11:25:21.926189058 -0400
+++ ./nordicsdk/drivers_nrf/uart/nrf_drv_uart.c	2019-04-07 12:06:44.012122825 -0400
@@ -817,9 +817,19 @@ __STATIC_INLINE void uart_irq_handler(NR
         }
         event.type                   = NRF_DRV_UART_EVT_ERROR;
         event.data.error.error_mask  = nrf_uart_errorsrc_get_and_clear(p_uart);
-        event.data.error.rxtx.bytes  = p_cb->rx_buffer_length;
+        event.data.error.rxtx.bytes  = p_cb->rx_counter;
         event.data.error.rxtx.p_data = p_cb->p_rx_buffer;
 
+        if (event.data.error.error_mask | NRF_UART_ERROR_OVERRUN_MASK)
+        {
+            nrf_uart_rxd_get(p_uart);
+            nrf_uart_rxd_get(p_uart);
+            nrf_uart_rxd_get(p_uart);
+            nrf_uart_rxd_get(p_uart);
+            nrf_uart_rxd_get(p_uart);
+            nrf_uart_rxd_get(p_uart);
+        }
+
         //abort transfer
         p_cb->rx_buffer_length = 0;
         p_cb->rx_secondary_buffer_length = 0;
