diff -ru nordicsdk-16.0.0-orig/components/libraries/cli/nrf_cli.c nordicsdk-16.0.0/components/libraries/cli/nrf_cli.c
--- nordicsdk-16.0.0-orig/components/libraries/cli/nrf_cli.c	2019-10-18 22:22:38.000000000 -0400
+++ nordicsdk-16.0.0/components/libraries/cli/nrf_cli.c	2020-02-15 11:04:10.642286203 -0500
@@ -3585,7 +3585,7 @@
     uint8_t utilization = (uint8_t)(max_util * 100ul / p_queue->size);
 
     nrf_cli_print(p_cli,
-                  "Lost logs: %u\nMax log queue utilization: %u%% [%u/%u]",
+                  "Lost logs: %" PRIu32 "\nMax log queue utilization: %u%% [%u/%u]",
                   p_cli->p_ctx->statistics.log_lost_cnt,
                   utilization,
                   max_util,
diff -ru nordicsdk-16.0.0-orig/components/libraries/cli/nrf_cli.h nordicsdk-16.0.0/components/libraries/cli/nrf_cli.h
--- nordicsdk-16.0.0-orig/components/libraries/cli/nrf_cli.h	2019-10-18 22:22:38.000000000 -0400
+++ nordicsdk-16.0.0/components/libraries/cli/nrf_cli.h	2020-02-15 11:00:14.784836079 -0500
@@ -571,6 +571,7 @@
  * @param[in] p_fmt Format string.
  * @param[in] ...   List of parameters to print.
  */
+__attribute__((format(printf, 3, 4)))
 void nrf_cli_fprintf(nrf_cli_t const *      p_cli,
                      nrf_cli_vt100_color_t  color,
                      char const *           p_fmt,
